exercise_3/ARecord.hpp

```diff
@@ -0,0 +1,20 @@
+//
+// Created by denys on 20.07.19.
+//
+
+#pragma once
+
+
+#include "IRecord.hpp"
+#include <list>
+
+class ARecord : public IRecord {
+private:
+    virtual std::list<std::pair<std::string, std::string>> _get_print_parameters() const = 0;
```

Comment:
An overcomplicated structure used.
I'd use a simple structure for that (containing two std::strings) and wrap it within std::vector.

Reply:
Like this ```std::vector<std::pair<std::string, std::string>>```?

Reply:
A bit better.  Better because std::vector is stored in a continuous are of memory thus is more efficient than a single list (usually).

Altough, still thinking about:
```
struct Parameter {
    std::string key;
    std::string value;
};
std::vector<Parameter> _get_print_parameters const = 0;
```